### PR TITLE
Schema changes detection with HostClient

### PR DESCRIPTION
**Pull request by gabrielperezs**
Related with the issue https://github.com/valyala/fasthttp/issues/244

### CODE REVIEW

File: http.go [hunk 1]
```diff
@@ -44,6 +44,9 @@ type Request struct {
 	keepBodyBuffer bool
 
 	isTLS bool
+
+	// To detect scheme changes in redirects
+	SchemaUpdate bool
```
Comment (erikdubbelboer) [inline http.go]: I would keep this private (`schemaUpdate`).

**Comment by erikdubbelboer** [comment]
Can you maybe add a test case for this to `client_test.go`?

File: http.go (hunk 1 above)
Comment (gabrielperezs) [inline http.go]: Yes, agree, this must be private I will change it

**Comment by gabrielperezs** [comment]
@erikdubbelboer test case done :)

File: client.go [hunk 2]
```diff
@@ -1047,6 +1062,22 @@ func (c *HostClient) doNonNilReqResp(req *Request, resp *Response) (bool, error)
 	// so the GC may reclaim these resources (e.g. response body).
 	resp.Reset()
 
+	// If we detected a redirect to another schema
+	if req.schemaUpdate {
+		if bytes.Equal(req.URI().Scheme(), strHTTPS) {
```
Comment (arbrix) [inline client.go]: Hi, great job.

Propose to simplify this `if` statement:
```
c.IsTLS = bytes.Equal(req.URI().Scheme(), strHTTPS)
c.Addr = addMissingPort(string(req.Host()), c.IsTLS)
c.addrIdx = 0
c.addrs = nil
```

Comment (gabrielperezs) [inline client.go]: Thanks @arbrix , included in the last commit.

**Comment by gabrielperezs** [comment]
Hello @valyala, did you has the change to check this pull request? Something to improve?

**Comment by erikdubbelboer** [comment]
This pull request has my approval, it has been merged and tested in my branch for months already.

**Comment by erikdubbelboer** [comment]
@gabrielperezs can you please rebase this on master and fix the merge conflicts. If not I will do it for you as I think this change is important. Thanks.

**Comment by erikdubbelboer** [comment]
@gabrielperezs there still seems to be merge conflicts.

**Comment by erikdubbelboer** [comment]
@gabrielperezs I fixed the merge conflicts and force pushed them to your fork. I hope you don't mind.

@kirillDanshin you can merge this if you agree with these changes.

**Comment by gabrielperezs** [comment]
@erikdubbelboer I don't mind. Thanks for it :+1: . I didn't have the time to check it.

**Comment by kirillDanshin** [comment]
@gabrielperezs @erikdubbelboer thanks!

**Pull request merged**